src/cswinrt/code_writers.h

```diff
@@ -5499,20 +5581,65 @@ private % AsInternal(InterfaceTag<%> _) => _default;
             bind([&](writer& w)
             {
                 bool has_base_type = !std::holds_alternative<object_type>(get_type_semantics(type.Extends()));
+                if (is_fast_abi_class(type))
+                {
+                    int hierarchy_index = get_class_hierarchy_index(type);
+                    if (hierarchy_index > 0 || !type.Flags().Sealed())
+                    {
+                        auto default_interface = get_default_interface(type);
+                        auto default_iface_method_count = 0;
+                        for_typedef(w, get_type_semantics(default_interface), [&](TypeDef iface)
+                        {
+                            default_iface_method_count = distance(iface.MethodList());
+                        });
+
+                        w.write(R"(
+protected unsafe % List<Func<IntPtr>> GetDefaultFactories()
+{
+var list = new List<Func<IntPtr>>();
+%
+return list;
+})",
+                        hierarchy_index == 0 ? "virtual" : "override",
+                        bind([&](writer& w)
+                        {
+                            for (auto i = 0; i < hierarchy_index; i++)
+                            {
+                                w.write("list.Add(() => (*(delegate* unmanaged[Stdcall]<IntPtr, IntPtr>**)_inner.ThisPtr)[%](_inner.ThisPtr));\n", 6 + default_iface_method_count + i);
```

Comment:
Would it be better if this was instead an array of ints and we used NativeObject to get the thisptr and then use the int we get from this list to do the offset.  That would mean we eliminate the need for the funcs.